spec.bs

```diff
@@ -721,32 +774,72 @@ To <dfn>create a request</dfn> given a [=URL=] |url|, a [=string=] <dfn for="cre
 
-To <dfn>fetch resource</dfn> given a [=URL=] |url|, a [=string=] |mimeType|, and an [=origin=]
-|origin| or null:
-1. Let |request| be the result of [=creating a request=] with |url|, [=create a request/accept=]
-  set to |mimeType|, and [=create a request/origin=] set to |origin|.
-1. Let |resource| be null.
-1. [=Fetch=] |request| with [=fetch/processResponseConsumeBody=] set to the following steps given a
-  [=response=] |response| and |responseBody|:
-  1. If |responseBody| is null or failure, set |resource| to failure and return.
+  1. If |responseBody| is null or failure, return false.
   1. Let |headers| be |response|'s [=response/header list=].
   1. If [=header list/getting a structured field value=] "X-Allow-Protected-Audience" from |headers| does not
-    return true, set |resource| to failure and return.
+    return true, return false.
   1. Let |headerMimeType| be the result of [=header list/extracting a MIME type=] from |headers|.
-  1. Set |resource| to failure and return, if any of the following conditions hold:
+  1. Return false if any of the following conditions hold:
     * |headerMimeType| is failure;
     * |mimeType| is "`text/javascript`" and |headerMimeType| is not a [=JavaScript MIME type=];
     * |mimeType| is "`application/json`" and |headerMimeType| is not a [=JSON MIME type=].
   1. Let |mimeTypeCharset| be |headerMimeType|'s [=MIME type/parameters=]["`charset`"].
-  1. Set |resource| to failure and return if any of the following conditions hold:
+  1. Return false if any of the following conditions hold:
     * If |mimeTypeCharset| does not [=map/exist=], or |mimeTypeCharset| is "utf-8", and
       |responseBody| is not [=UTF-8=] encoded;
     * If |mimeTypeCharset| is "us-ascii", and |responseBody| is not [=ascii string=].
-  1. Set |resource| to |responseBody|.
-1. Wait for |resource| to be set.
-1. Return |resource|.
+  1. Return true.
+</div>
+
+<div algorithm>
+To <dfn>fetch script</dfn> given a [=URL=] |url|:
+
+  1. Let |request| be the result of [=creating a request=] with |url|, "`text/javascript`", and
+    null.
+  1. Let |script| be null.
+  1. [=Fetch=] |request| with [=fetch/processResponseConsumeBody=] set to the following steps given
+    a [=response=] |response| and |responseBody|:
+    1. If [=validate fetching response=] with |response|, |responseBody| and "`text/javascript`"
+      returns false, set |script| to failure and return.
+    1. Set |script| to |responseBody|.
+  1. Wait for |script| to be set.
+  1. Return |script|.
+</div>
 
+<div algorithm>
+To <dfn>fetch trusted signals</dfn> given a [=URL=] |url|, and a [=boolean=] |isBiddingSignal|:
+
+  1. Let |request| be the result of [=creating a request=] with |url|, "`application/json`", and
+    null.
+  1. Let |signals| be null.
+  1. Let |dataVersion| be null.
+  1. [=Fetch=] |request| with [=fetch/processResponseConsumeBody=] set to the following steps given
+    a [=response=] |response| and |responseBody|:

```

Comment:
What is the type of |responseBody|?

Reply:
It's "null, failure, or a byte sequence" according to the definition of [processResponseConsumeBody](https://fetch.spec.whatwg.org/#process-response-end-of-body).  I guess since it can be one of three things with different types, it doesn't get an explicit type declaration.